### PR TITLE
MPDX-7136-Contacts-Tasks-Mass-Selection-Hook

### PR DESCRIPTION
**Pull request by TheNoodleMoose**
This PR is for the creation of the ```useMassSelection``` hook, as well as adding the UI for the mass actions buttons. Connecting everything to the actions buttons and mass actions endpoint will happen in a separate PR(once the endpoint is available). Based the logic on the comments @OzzieOrca left in the [jira ticket](https://jira.cru.org/browse/MPDX-7136). Some of the logic can probably be cleaned up, but I at least got it working the way it should! Comments and questions appreciated 🙂

### CODE REVIEW

**Review by OzzieOrca — CHANGES_REQUESTED** [review]
Looks like a really clean way to do this! I like that you got pretty much all the code to live inside the hook :)

File: pages/accountLists/[accountListId]/contacts/[[...contactId]].page.tsx [hunk 1]
```diff
@@ -78,15 +87,29 @@ const ContactsPage: React.FC = () => {
...
+  const handleCheckOneContact = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    contactId: string,
+  ): void => {
+    toggleSelectionById(contactId);
+  };
+
+  const handleCheckAllContacts = (): void => {
+    toggleSelectAll();
   };
```
[the remark this answers is not in the document]
Comment (TheNoodleMoose) [reply pages/accountLists/[accountListId]/contacts/[[...contactId]].page.tsx:109]: Yeah, I agree. Will update!

File: src/hooks/useMassSelection.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        break;
+    }
+  };
+
+  const isRowChecked = (id: string) =>
+    (selectionType === ListHeaderCheckBoxState.partial && ids.includes(id)) ||
+    (selectionType === ListHeaderCheckBoxState.checked &&
+      !reverseIds.includes(id));
+
+  return {
```
[the remark this answers is not in the document]
Comment (TheNoodleMoose) [reply src/hooks/useMassSelection.ts:94]: You know I had it returning those and then was like "Oh I don't need those at the moment, let me just remove them"  haha. I'll add them back.

File: src/hooks/__tests__/useMassSelection.test.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+describe('useMassSelection', () => {
+  it('should return selectionType', () => {
+    const { result } = renderHook(() => useMassSelection(10), {});
+
+    expect(result.current.selectionType).toEqual(
+      ListHeaderCheckBoxState.unchecked,
+    );
+  });
+
+  //#region ToggleSelectionById Tests
```
[the remark this answers is not in the document]
Comment (TheNoodleMoose) [reply src/hooks/__tests__/useMassSelection.test.ts:15]: Ah I forgot I left this in when I first tested regions. Maybe a combination of both? The ```describe``` adds organization in the tests suite and ``` the ```region```s can provide organization in the IDE.

File: src/hooks/useMassSelection.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      setSelectionType(ListHeaderCheckBoxState.checked);
+      setIds([]);
+    }
+  }, [ids, reverseIds]);
+
+  const toggleSelectionById = (id: string) => {
+    switch (selectionType) {
+      case ListHeaderCheckBoxState.partial:
+        const currentIndex = ids.indexOf(id);
+        if (currentIndex !== -1) {
```
[the remark this answers is not in the document]
Comment (TheNoodleMoose) [reply src/hooks/useMassSelection.ts:44]: Yeah, I like that better. I'll switch to that.

File: src/components/Shared/Header/ListHeader.tsx [hunk 5]
```diff
@@ -154,8 +167,56 @@ export const ListHeader: React.FC<ListHeaderProps> = ({
       {page === 'contact' ? (
         <>
           <Hidden smDown>
-            {/*TODO: Replace this with Actions Dropdown*/}
-            <PlaceholderActionsDropdown />
```
[the remark this answers is not in the document]
Comment (TheNoodleMoose) [reply src/components/Shared/Header/ListHeader.tsx:158]: I renamed it from ```PlaceholderActionsDropdown``` to ```ActionsButton```.

File: src/hooks/__tests__/useMassSelection.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (OzzieOrca) [reply src/hooks/__tests__/useMassSelection.test.ts:17]: Sure :)

**Review by OzzieOrca — APPROVED** [review]
🎉

**Pull request merged**